introspection/org.freedesktop.NetworkManager.Device.Wireguard.xml

```diff
@@ -0,0 +1,48 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<node name="/">
+  <!--
+      org.freedesktop.NetworkManager.Device.Wireguard:
+      @short_description: WireGuard Device
+
+  -->
+  <interface name="org.freedesktop.NetworkManager.Device.Wireguard">
+
+    <!--
+        ListenPort:
+
+        Local UDP listening port.
+    -->
+    <property name="ListenPort" type="u" access="read"/>
+
+    <!--
+        Fwmark:
+
+        Firewall mark.
+    -->
+    <property name="Fwmark" type="u" access="read"/>
+
+    <!--
+        PrivateKey:
+
+        32-byte private WireGuard key.
+    -->
+    <property name="PrivateKey" type="v" access="read"/>
```

Comment:
"v" signature ? Wouldn't "ay" make more sense for a byte array?

Also, the PrivateKey probably shouldn't be visible on the D-Bus API, since it has not access control and is visible to all local users. Is it necessary to have this?

Reply:
Well, it still doesn't have to be base64 on D-Bus API -- it may make more sense to encode it in nmcli and possibly the keyfile plugin. @thom311, what do you think?

Reply:
I also would think that the private key is either a binary array (ay) or a string (s).

Maybe string is nicer, because then we can have there the base64 key directly. And if we some day in the future a different format (but why would we?), then we possibly could even extend the string with a "prefix:" (since ':' cannot be in plain base64).

Reply:
If we settle on base64, then it the documentation comment should state that it's base64-encoded.

Reply:
> If we settle on base64, then it the documentation comment should state that it's base64-encoded.
That's fine.

Also, I hadn't looked into the access control for the D-Bus API yet. If indeed it's readable by all non-root users then of course we can't have the private key there :)